modules/supervisord.nix

```diff
@@ -0,0 +1,177 @@
+{ pkgs, lib, config, ... }:
+let
+  inherit (lib) types;
+
+  cfg = config.supervisord;
+
+  format = pkgs.formats.ini {};
+
+  programType = types.submodule ({ name, config, ... }: {
+    options = {
+      enable = lib.mkOption {
+        description = ''
+          Whether to enable this program.
+        '';
+        type = types.bool;
+        default = true;
+      };
+      command = lib.mkOption {
+        description = ''
+          The command that will be run as the service's main process.
+        '';
+        type = types.str;
+        default = toString (pkgs.writeShellScript "${name}-script.sh" config.script);
+      };
+      script = lib.mkOption {
+        description = ''
+          Shell commands executed as the service's main process.
+        '';
+        type = types.lines;
+        default = "";
+      };
+      path = lib.mkOption {
+        description = ''
+          Packages added to the service's PATH environment variable.
+        '';
+        type = types.listOf (types.either types.package types.str);
+        default = [];
+      };
+      autoRestart = lib.mkOption {
+        description = ''
+          Whether to automatically restart the process if it exits.
+        '';
+        type = types.either types.bool (types.enum [ "false" "true" "unexpected" ]);
+        default = "unexpected";
+      };
+      environment = lib.mkOption {
+        description = ''
+          Environment variables passed to the service's process.
+        '';
+        type = types.attrsOf types.str;
+        default = {
+          PATH = lib.makeBinPath config.path;
```

Comment:
Defining PATH as default value can result in confusing behaviour. E.g. if I set `path` to a list with pkg1 and set anything in `environment`, pkg1 will not be available.

I would prefer setting PATH independent of `environment` and have an assertion to protect overrides like this.